File: manuscripts/scripts/figure1_discovery_rates.R
# Figure 1: Discovery Rates Bar Chart
# PAR(2) Circadian Gating Analysis
# Usage: Rscript figure1_discovery_rates.R

library(ggplot2)
library(dplyr)

# Read the data
data <- read.csv("figure1_discovery_rates.csv")

# Create the plot
p <- ggplot(data, aes(x = reorder(Tissue, -as.numeric(Rate)), y = as.numeric(Rate))) +
  geom_bar(stat = "identity", fill = "#22d3ee", width = 0.7) +
  geom_text(aes(label = paste0(Significant, "/", Total)), 
            vjust = -0.5, size = 3, color = "gray30") +
  labs(
    x = "Tissue",
    y = "Discovery Rate (%)",
    title = "Circadian Gating Discovery Rates Across Mouse Tissues",
    subtitle = "Significant PAR(2) clock-target pairs at p < 0.05"
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(face = "bold", size = 14),
    plot.subtitle = element_text(color = "gray50", size = 10),
    axis.text.x = element_text(angle = 45, hjust = 1, size = 10),
    axis.title = element_text(size = 11),
    panel.grid.major.x = element_blank(),
    panel.grid.minor = element_blank()
  ) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))

# Save the figure
ggsave("figure1_discovery_rates.pdf", p, width = 8, height = 6, dpi = 300)
ggsave("figure1_discovery_rates.png", p, width = 8, height = 6, dpi = 300)

cat("Figure 1 saved as figure1_discovery_rates.pdf and .png\n")

File: manuscripts/scripts/figure1_discovery_rates.py
#!/usr/bin/env python3
"""
Figure 1: Discovery Rates Bar Chart
PAR(2) Circadian Gating Analysis
Usage: python figure1_discovery_rates.py
"""

import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

# Read the data
data = pd.read_csv("figure1_discovery_rates.csv")
data = data.sort_values('Rate', ascending=False)

# Create figure
fig, ax = plt.subplots(figsize=(10, 6))

# Create bar chart
bars = ax.bar(data['Tissue'], data['Rate'].astype(float), 
              color='#22d3ee', edgecolor='#0e7490', linewidth=1)

# Add value labels on bars
for bar, sig, total in zip(bars, data['Significant'], data['Total']):
    height = bar.get_height()
    ax.annotate(f'{sig}/{total}',
                xy=(bar.get_x() + bar.get_width() / 2, height),
                xytext=(0, 3),
                textcoords="offset points",
                ha='center', va='bottom', fontsize=9, color='gray')

# Styling
ax.set_xlabel('Tissue', fontsize=12)
ax.set_ylabel('Discovery Rate (%)', fontsize=12)
ax.set_title('Circadian Gating Discovery Rates Across Mouse Tissues', 
             fontsize=14, fontweight='bold')
ax.set_ylim(0, max(data['Rate'].astype(float)) * 1.15)

# Rotate x labels
plt.xticks(rotation=45, ha='right')

# Remove top and right spines
ax.spines['top'].set_visible(False)
ax.spines['right'].set_visible(False)

# Grid
ax.yaxis.grid(True, linestyle='--', alpha=0.3)
ax.set_axisbelow(True)

plt.tight_layout()

# Save
plt.savefig('figure1_discovery_rates.pdf', dpi=300, bbox_inches='tight')
plt.savefig('figure1_discovery_rates.png', dpi=300, bbox_inches='tight')
print("Figure 1 saved as figure1_discovery_rates.pdf and .png")
